vignettes/vignette_best_practices.Rmd

```diff
@@ -0,0 +1,198 @@
+
+```
+# Introduction
+
+This is the introduction of my mignette.
+
+## Required Packages
+
+    `r ''````{r, warning=FALSE, message=FALSE}
+    library(admiral)
+    # <all packages used in the vignette>
+    ```
+```
+
+The `warning = FALSE` and `message = FALSE` options are there to prevent the usual messages:
+
+<span style="color: red;font-family: textmate; font-size = 12pt;" >Attaching package: 'xxxx'</span>
+
+<span style="color: red;font-family: textmate; font-size = 12pt;">The following objects are masked from 'package:yyyyy':</span>
+
+<span style="color: red;font-family: textmate; font-size = 12pt;">  fun1, fun2</span>
+
+### Conventions
+
+#### General Conventions
+
++ Any variable name, dataset name, function, argument name must be quoted with backticks: e.g. '`'ADSL'`', '`'dm'`', '`'ADT'`'
+
++ Functions must also end with `()`.
+
++ All the codes must be executed and the output result displayed once the code is executed. Use:
+ 
+
+      `r ''````{r}
+      #<code>
+      ```
+
++ Any output created must clearly show what the function has derived. 
+It should at least show the variables/records used as input to the function and the derived 
+variables/records.
+
+#### Conventions for ADaM Workflow
+
+For vignettes describing an ADaM workflow, 
+
++ the second section will summarize the programming workflow. The first sub-section within this 
+workflow will always describe the data read to demonstrate the use of the admiral functions,
+
++ Each sub-section within the programming workflow should be tagged (e.g. [Step1] (#step)), so that 
+the user can go on the relevant section from the programming workflow (in addition to the Table of
+contents). Don’t use a tag with a number but use a meaningful name (e.g. do not use `(#link1)`,
+but use `(#this_action)`)
+
++ the last section should link to a template script. 
+
+```
+# Programming Workflow
+
+* [Read in Data](#readdata)
+* [Derive/Impute End and Start Analysis Date/time and Relative Day](#datetime)
+* ...
+* [Assign `ASEQ`](#aseq)
+
+# Another Section
+
+# Example Script
+ADaM | Sample Code
+---- | --------------
+ADxx | [ad_adxx.R](https://github.com/Roche-GSK/admiral/blob/master/inst/templates/ad_adxx.R){target="_blank"}
+
+
+```
+
++ In order to clearly demonstrate the purpose of a function, 
+  + the code used to derive the variable/record ...
+
+```{r, include=FALSE}
+library(dplyr)
+library(lubridate)
+library(DT)
+library(admiral)
+
+data(dm)
+data(ds)
+```
+
+
+```{r, eval=FALSE, echo=TRUE}
+adsl <- dm %>%
+  derive_disposition_dt(
+    dataset_ds = ds,
+    new_var = EOSDT,
+    dtc = DSSTDTC,
+    filter = DSCAT == "DISPOSITION EVENT" & DSDECOD != "SCREEN FAILURE",
+    date_imputation = NULL
+  )
+
+
+```
+
+
+  + ... must be followed by the output dataset, formatted using `dataset_vignette()`...
+  
+  
+      
+      `r ''````{r}
+        dataset_vignette(adsl, 
+                        display_vars=vars(USUBJID, RFSTDTC, RFENDTC, DTHDTC, ARMCD, EOSDT),
```

Comment:
I would remove `RFSTDTC`, `RFENDTC`, `DTHDTC` and add `DSSTDTC` and `DSCAT` and `DSDECOD` as those variables are relevant to the call to the function `derive_disposition_dt()` above.

Reply:
Example updated to show the `derive_vars_dt` because the `derive_disposition_xx` functions do not keep the variables (DSDECOD, DSSACT,...) used to derive the disposition variable.